src/staking/block_validator.cpp

```diff
@@ -84,30 +89,28 @@ class BlockValidatorImpl : public AbstractBlockValidator {
     // read + check height
     check = script_sig.GetOp(it, op, buf);
     if (!check || (buf.empty() && op != OP_0)) {
-      result.AddError(Error::NO_BLOCK_HEIGHT);
-      result.AddError(Error::NO_SNAPSHOT_HASH);
```

Comment:
Any reason to have this gone?

Reply:
It's not gone, it is still checked in https://github.com/dtr-org/unit-e/pull/1098/files/5080518baddb6ae19077366b27f762d5647f5e76#diff-3fa6b9dd4cec73397620af53de60d1ebL105

This pull request changes the notion of the `BlockValidationResult` to no longer accumulate a set of errors but only report the first error encountered. This line adds two errors (as there is no snapshot hash it there is already no block height) to the result set. The new version returns with the first error condition encountered here: no block height.

The presence of the snapshot hash is checked in line 108.

Reply:
ah right, cool then.